sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/UcfgInstructionFactory.cs

```diff
@@ -415,6 +416,26 @@ private IEnumerable<Instruction> ProcessAssignmentExpression(AssignmentExpressio
 
             var rightExpression = expressionService.GetExpression(assignmentExpression.Right);
 
+            // Handle the special add assignment (+=)
+            // In this block we will only handle the concatenation part, not the assignment (that will be handled right after)
+            if (assignmentExpression.IsKind(SyntaxKind.AddAssignmentExpression))
```

Comment:
+= is a special case of the assignment but to keep code quite simple, I have tried to handle only the + part and let the assignment as it was. Basically `a += b` is handled as first `a + b` and then the classic assignment `a = result`.

Reply:
Sounds reasonable. I think it would be worth extracting this code to a separate method.